### PR TITLE
Roman bordulyak branch

### PR DESCRIPTION
**Pull request by Edem404**
My homework for team work

### CODE REVIEW

File: group.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+class Group
+{
+public:
+    Group()
+    {
+    }
+
+    float sort(float arr[],int count)
```
Comment (kefirzmolokom) [inline group.h:9]: do not clear what this method for

Comment (kefirzmolokom) [inline group.h]: 'float sort(float arr[],int count)'
do not clear what this method for

File: laba5_test.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#include "student.h"
+#include "group.h"
+
+int main()
+{
+
+    int list_count;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'int list_count;'
list of what

File: student.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include <vector>
+
+using namespace std;
+
+class Student
+{
+public:
+    string name;
+    vector<int> marks;
+    float average;
```
Comment (kefirzmolokom) [inline student.h]: 'float average;'
pls rename this variable to make it clear what it is for

File: student.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    Student()
+    {
+        average = 0;
```
Comment (kefirzmolokom) [inline student.h]: 'average = 0;'
pls rename this variable to make it clear what it is for

File: student.h [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    int marks_output()
+    {
+        for (int i = 0; i < marks.size(); i++)
+        {
+            cout << marks[i] << ", ";
+        }
+        cout << "\t\t";
+    }
+
+    float average_create()
```
Comment (kefirzmolokom) [inline student.h]: 'float average_create()'
do not clear what this method for

File: student.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        for (int i = 0; i < marks.size(); i++)
+        {
+            average = average + marks[i];
```
Comment (kefirzmolokom) [inline student.h]: 'average = average + marks[i];'
pls rename "average" variable to make it clear what it is for

File: student.h [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        average = average / marks.size();
```
Comment (kefirzmolokom) [inline student.h]: 'average = average / marks.size();'
pls rename "average" variable to make it clear what it is for

File: student.h [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        return average;
```
Comment (kefirzmolokom) [inline student.h]: 'return average;'
pls rename "average" variable to make it clear what it is for

File: laba5_test.cpp [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    cin >> list_count;
+    vector<Student> list;
+    for (int index = 0; index < list_count; index++)
+    {
+        list.push_back(Student());
+    }
+
+    for (int i = 0; i < list_count; i++)
+    {
+        int length;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'int length;'
pls rename this variable to make it clear what it is for

File: laba5_test.cpp [hunk 10]
```diff
@@ -0,0 +1,81 @@
+#include "student.h"
+#include "group.h"
+
+int main()
+{
+
+    int list_count;
+    cout << "how many students in group?\t";
+    cin >> list_count;
+    vector<Student> list;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'vector<Student> list;'
it would be better to give it proper name like "list_of_students"

File: laba5_test.cpp [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        list.push_back(Student());
+    }
+
+    for (int i = 0; i < list_count; i++)
+    {
+        int length;
+        cout << "name of student: ";
+        cin >> list[i].name;
+        cout << "how many marks " << list[i].name << " have?\t" << endl;
+        cin >> length;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'cin >> length;'
pls rename this variable to make it clear what it is for

File: laba5_test.cpp [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        list[i].marks_arr_create(length);
+        list[i].average_create();
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'list[i].average_create();'
do not clear what "average_create()" is for

File: group.h (hunk 1 above)
Comment (Edem404) [reply group.h]: Okay,this copied from inet =)

File: laba5_test.cpp [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    Group ir_12;
+
+    float arr_average [list_count];
+    for (int i = 0; i < list_count; i++)
+    {
+
+        arr_average[i] = list[i].average;
+    }
+
+    ir_12.sort(arr_average, list_count);
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'ir_12.sort(arr_average, list_count);'
do not clear what "sort(arr_average, list_count);" is for;

File: laba5_test.cpp [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+    cout << "____________________________________________________" << endl;
+    cout << "do you wanna see the top of students?";
+    cout << " | 1-yes\t2-no | ";
+    int choise;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'int choise;'
pls rename this variable to make it clear what it is for

File: laba5_test.cpp [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+    cin >> choise;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'cin >> choise;'
pls rename "choise" variable to make it clear what it is for

File: laba5_test.cpp [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    cout << "____________________________________________________" << endl;
+
+    if (choise == 1)
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'if (choise == 1)'
pls rename "choise" variable to make it clear what it is for

File: group.h [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        int i_sort;
+        for (i_sort = 0; i_sort < count; i_sort++)
+        {
+            int first = i_sort;
+
+            for (int h = i_sort + 1; h < count; h++)
```
Comment (kefirzmolokom) [inline group.h]: 'for (int h = i_sort + 1; h < count; h++)'
pls give a proper name to "h" variable

File: laba5_test.cpp [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+    for (int i = 0; i < list_count; i++)
+    {
+        cout << list[i].name << " \t";
+        list[i].marks_output();
+        cout << list[i].average << endl;
+    }
+
+    Group ir_12;
+
+    float arr_average [list_count];
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'float arr_average [list_count];'
do not clear what this variable is for

File: laba5_test.cpp [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+    for (int i = 0; i < list_count; i++)
+    {
+
+        arr_average[i] = list[i].average;
```
Comment (kefirzmolokom) [inline laba5_test.cpp]: 'arr_average[i] = list[i].average;'
pls rename "arr_average" variable to make it clear what it is for

**Review by kefirzmolokom — CHANGES_REQUESTED** [review]
its better to rename "laba5_test.cpp" to "laba5.cpp"

File: group.h [hunk 20]
```diff
@@ -6,20 +6,20 @@ class Group
     {
     }
 
-    float sort(float arr[],int count)
```
Comment (kefirzmolokom) [inline group.h]: function have return type float but aint return nothing

File: group.h [hunk 21]
```diff
@@ -6,20 +6,20 @@ class Group
...
+    float make_top_of_students(float arr_avarage_of_marks[],int num_of_students)
```
Comment (kefirzmolokom) [inline group.h]: 'float make_top_of_students(float arr_avarage_of_marks[],int num_of_students)'
function have return type "float" but aint return nothing

File: student.h [hunk 22]
```diff
@@ -9,16 +9,16 @@ class Student
...
+    float avarage_of_marks;
 
     Student()
     {
-        average = 0;
+        avarage_of_marks = 0;
     }
 
-    int marks_arr_create(int length)
+    int marks_arr_create(int num_of_marks)
```
Comment (kefirzmolokom) [inline student.h]: 'int marks_arr_create(int num_of_marks)'
function have return type "int" but aint return nothing

**Review by kefirzmolokom — COMMENTED** [review]
need some improvement

**Review by kefirzmolokom — APPROVED** [review]

**Review by kefirzmolokom — APPROVED** [review]
good job

File: laba_5.cpp [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+    for (int i = 0; i < num_of_students; i++)
+    {
+        cout << list_of_students[i].name << " \t";
+        list_of_students[i].marks_output();
+        cout << list_of_students[i].average_of_marks << endl;
+    }
+
+    Group ir_12;
+
+    float arr_average_of_marks [num_of_students];
```
Comment (kefirzmolokom) [inline laba_5.cpp:41]: 'float arr_average_of_marks [num_of_students];'
return type is "float" but function aint return nothing

**Review by kefirzmolokom — COMMENTED** [review]
noice

**Pull request merged**